runtime/permissions/lib.rs

```diff
@@ -688,6 +687,28 @@ impl Descriptor for WriteDescriptor {
   }
 }
 
+/// Modify `fqdn!` to handle domain names ending with a dot
+/// Handle parsing error
+macro_rules! fqdn {
+    ($($args:expr),*) => {{
+        #[allow(unused_mut)]
+        let mut str = std::string::String::new();
+        $( str += "."; str += $args; )*
+
+
+        let fqdn_string = if str.ends_with('.') {
+            &str[1..str.len() - 1]
+        } else {
+            &str[1..]
+        };
+
+        match fqdn_string.parse::<$crate::FQDN>() {
+            Ok(fqdn) => fqdn,
+            Err(_) => $crate::FQDN::default(),
```

Comment:
This shouldn't return `FQDN::default` when failing to parse -- the callsites need to be checking the errors instead.

Reply:
@mmastrac 
it's in parsing the domain, this the only way to prevent panic by convert it to default, so it will never connect to domain, it will return error when it tries to connect to default domain (.) ,
if u have some suggestion for what should i can update it tell me.

Reply:
Hi, it's done
I have implemented all the changes asked of me.